pkg/platform/abstract/platform.go

```diff
@@ -481,6 +481,17 @@ func (ap *Platform) validateFunctionName(functionConfig *functionconfig.Config)
 	return nil
 }
 
+func (ap *Platform) AutoFixConfiguration(ctx context.Context, err error, functionConfig *functionconfig.Config) bool {
```

Comment:
My idea was to check validation only on `kube` and `abstract` levels if platform is kube, and `local` and `abstract` if platform is local. To follow the same logic we do with enrichment and validation. 

But I agree that `functionconfig.AutoFixConfig` approach is simpler.

Reply:
@TomerShor then with `functionconfig.AutoFixConfig` approach we will have to pass `ctx` and `logger` right to `Config`' and log from there. Do we really want to do like this? Because now we don't have any ctx and logger usages in Config's methods

Reply:
I don't want to introducer logger and context to the functionconfig package, so we can leave it here.